src/Annotation/Controller/RequestParameters.php

```diff
@@ -31,6 +31,24 @@ public function __construct($parameters)
 			$parameters = [$parameters];
 		}
 
+		$this->validateUniqueNames($parameters);
+
+		$this->parameters = $parameters;
+	}
+
+	/**
+	 * @return RequestParameter[]
+	 */
+	public function getParameters(): array
+	{
+		return $this->parameters;
+	}
+
+	/**
+	 * @param RequestParameter[] $parameters
+	 */
+	private function validateUniqueNames(array $parameters): void
+	{
```

Comment:
Decoupling complex validation code that does not modify/work with values makes the class more readable